src/components/structures/auth/ForgotPassword.tsx

```diff
@@ -154,94 +122,77 @@ export default class ForgotPassword extends React.Component<IProps, IState> {
         });
     }
 
-    public submitPasswordReset(email: string, password: string, logoutDevices = true): void {
-        this.setState({
-            phase: Phase.SendingEmail,
-        });
-        this.reset = new PasswordReset(this.props.serverConfig.hsUrl, this.props.serverConfig.isUrl);
-        this.reset.resetPassword(email, password, logoutDevices).then(() => {
-            this.setState({
-                phase: Phase.EmailSent,
-            });
-        }, (err) => {
-            this.showErrorDialog(_t('Failed to send email') + ": " + err.message);
-            this.setState({
-                phase: Phase.Forgot,
-            });
-        });
-    }
+    private async onPhaseEmailInputSubmit() {
+        this.phase = Phase.SendingEmail;
 
-    private onVerify = async (ev: React.MouseEvent): Promise<void> => {
-        ev.preventDefault();
-        if (!this.reset) {
-            logger.error("onVerify called before submitPasswordReset!");
+        if (await this.sendVerificationMail()) {
+            this.phase = Phase.EmailSent;
             return;
         }
-        if (this.state.currentHttpRequest) return;
 
+        this.phase = Phase.EnterEmail;
+    }
+
+    private sendVerificationMail = async (): Promise<boolean> => {
         try {
-            await this.handleHttpRequest(this.reset.checkEmailLinkClicked());
-            this.setState({ phase: Phase.Done });
-        } catch (err) {
-            this.showErrorDialog(err.message);
+            await this.reset.requestResetToken(this.state.email);
+            return true;
+        } catch (err: any) {
```

Comment:
proper types please (applies throughout)

Reply:
huh, that's both interesting and annoying. We might consider using `unknown` instead, but that's a conversation for another PR or at least another time :P